### PR TITLE
[systemtest] Remove unnecessary external listeners

### PR DESCRIPTION
**Pull request by im-konge**
Signed-off-by: Lukas Kral <lkral@redhat.com>

### Type of change

- Enhancement

### Description

This PR will remove unnecessary external listeners from tests where are not really needed.

### Checklist

- [x] Remove unnecessary external listeners
- [x] Make sure all tests pass

### CODE REVIEW

**Comment by im-konge** [comment]
@stanlyDoge run tests test_only profile=internalclients testcase=ConnectST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=internalclients testcase=ConnectST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=internalclients testcase=ConnectST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=internalclients testcase=ConnectST,UserST

**Comment by im-konge** [comment]
@strimzi-ci run tests test_only profile=regression testcase=UserST

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java [hunk 1]
```diff
@@ -157,43 +140,38 @@ void testKafkaConnectWithFileSinkPlugin() throws Exception {
...
-                            .endKafkaListenerExternalNodePort()
                         .endListeners()
                     .endKafka()
                 .endSpec()
                 .done();
 
-        KafkaUserResource.scramShaUser(CLUSTER_NAME, USER_NAME).done();
-        KafkaUserUtils.waitForKafkaUserCreation(USER_NAME);
+        KafkaUser kafkaUser = KafkaUserResource.scramShaUser(CLUSTER_NAME, USER_NAME).done();
+        SecretUtils.waitForSecretReady(USER_NAME);
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: This is already part of default check for user creation.

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java [hunk 2]
```diff
@@ -245,21 +229,9 @@ void testKafkaConnectWithPlainAndScramShaAuthentication() throws InterruptedExce
...
-                    .editKafka()
-                    .editListeners()
-                        .withNewKafkaListenerExternalNodePort()
-                            .withTls(false)
-                        .endKafkaListenerExternalNodePort()
-                    .endListeners()
-                    .endKafka()
-                .endSpec()
-                .done();
+    void testKafkaConnectAndConnectorFileSinkPlugin() {
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: Missing internal_clients tag

File: systemtest/src/test/java/io/strimzi/systemtest/UserST.java [hunk 3]
```diff
@@ -201,15 +201,7 @@ void createBigAmountOfUsers(String typeOfUser) {
...
-        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1)
-            .editSpec()
-                .editKafka()
-                    .editListeners()
-                        .withNewKafkaListenerExternalNodePort()
-                        .endKafkaListenerExternalNodePort()
-                    .endListeners()
-                .endKafka()
-            .endSpec().done();
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, 1, 1).done();
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/UserST.java]: I think you can now remove some `NODEPORT_SUPPORTED` tags from`UserST`

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java (hunk 2 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: yep my bad

File: systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java (hunk 1 above)
Comment (im-konge) [reply systemtest/src/test/java/io/strimzi/systemtest/ConnectST.java]: my bad :)

**Comment by im-konge** [comment]
@Frawless thank you for nits, ready for merge now

**Comment by scholzj** [comment]
@im-konge The Travis build seems to be failing on checkstyle ... can you fix it first?

**Comment by im-konge** [comment]
@scholzj my bad, fixed..

**Comment by scholzj** [comment]
Thanks.

**Pull request merged**